apps/examples/testcase/le_tc/filesystem/tc_fs_mops.c

```diff
@@ -0,0 +1,455 @@
+}
+
+static void tc_fs_mops_rmdir(const char *filesystemtype)
+{
+	int ret = -1;
+
+	if (strcmp(filesystemtype, "tmpfs") == 0) {
+		ret = rmdir(TMPFS_TEST_MOUNTPOINT"/dir");
+	} else if (strcmp(filesystemtype, "romfs") == 0) {
+		return;
+	} else if (strcmp(filesystemtype, "smartfs") == 0) {
+		ret = rmdir(SMARTFS_TEST_MOUNTPOINT"/dir");
+	}
+
+	TC_ASSERT_EQ("rmdir", ret, OK);	
+}
+
+static void tc_fs_mops_stat(const char *filesystemtype)
+{
+	int ret = -1;
+	struct stat st;
+	char filename[100] = "/rom/init.d/rcS";
+
+	if (strcmp(filesystemtype, "tmpfs") == 0) {
+		ret = stat(TMPFS_TEST_MOUNTPOINT"/test", &st);
+	} else if (strcmp(filesystemtype, "romfs") == 0) {
+		ret = stat(filename, &st);
+	} else if (strcmp(filesystemtype, "smartfs") == 0) {
+		ret = stat(SMARTFS_TEST_MOUNTPOINT"/test", &st);
+	}
+
+	TC_ASSERT_EQ("stat", ret, OK);	
+}
+
+static void tc_fs_mops_rename(const char *filesystemtype)
+{
+	int ret = -1;
+
+	if (strcmp(filesystemtype, "tmpfs") == 0) {
+		ret = rename(TMPFS_TEST_MOUNTPOINT"/test", TMPFS_TEST_MOUNTPOINT"/test_new");
+	} else if (strcmp(filesystemtype, "romfs") == 0) {
+		return;
+	} else if (strcmp(filesystemtype, "smartfs") == 0) {
+		ret = rename(SMARTFS_TEST_MOUNTPOINT"/test", SMARTFS_TEST_MOUNTPOINT"/test_new");
+	}
+
+	TC_ASSERT_EQ("rename", ret, OK);
+}
+
+static void tc_fs_mops_statfs(const char *filesystemtype)
+{
+	int ret = -1;
+	struct statfs fs;
+
+	if (strcmp(filesystemtype, "tmpfs") == 0) {
+		ret = statfs(TMPFS_TEST_MOUNTPOINT, &fs);
+	} else if (strcmp(filesystemtype, "romfs") == 0) {
+		ret = statfs("/rom", &fs);
+	} else if (strcmp(filesystemtype, "smartfs") == 0) {
+		ret = statfs(SMARTFS_TEST_MOUNTPOINT, &fs);
+	}
+
+	TC_ASSERT_EQ("statfs", ret, OK);
+}
+
+static void tc_fs_mops_unlink(const char *filesystemtype)
+{
+	int ret = -1;
+
+	if (strcmp(filesystemtype, "tmpfs") == 0) {
+		ret = unlink(TMPFS_TEST_MOUNTPOINT"/test_new");
+	} else if (strcmp(filesystemtype, "romfs") == 0) {
+		return;
+	} else if (strcmp(filesystemtype, "smartfs") == 0) {
+		ret = unlink(SMARTFS_TEST_MOUNTPOINT"/test_new");
+	}
+
+	TC_ASSERT_EQ("unlink", ret, OK);
+}
+
+static void tc_fs_mops_umount(const char *filesystemtype)
+{
+	int ret = -1;
+
+	if (strcmp(filesystemtype, "tmpfs") == 0) {
+		ret = umount(TMPFS_TEST_MOUNTPOINT);
+	} else if (strcmp(filesystemtype, "romfs") == 0) {
+		ret = umount("/rom");
+	} else if (strcmp(filesystemtype, "smartfs") == 0) {
+		ret = umount(SMARTFS_TEST_MOUNTPOINT);
+	}
+
+	TC_ASSERT_EQ("umount", ret, OK);
+}
+
+void tc_fs_mops_testcase(const char *filesystemtype)
+{
+	DIR *dir = NULL;
+	int fd1 = -1;
+
+	tc_fs_mops_mount(filesystemtype);
+	tc_fs_mops_open(filesystemtype, &fd1);
+	tc_fs_mops_write(filesystemtype, fd1);
+	tc_fs_mops_sync(filesystemtype, fd1);
+	tc_fs_mops_lseek(filesystemtype, fd1);
+	tc_fs_mops_read(filesystemtype, fd1);
+	tc_fs_mops_ioctl(filesystemtype, fd1);
+	tc_fs_mops_dup(fd1);
+	tc_fs_mops_fstat(filesystemtype, fd1);
+	close(fd1);
```

Comment:
In this format, if one tc fails, all the subsequent tc's also fail. I think its better to open & close everytime.

Reply:
I think that is also true.
However, since this test is done on a mount point operation basis, I think it is not wrong either.